designs/isolate-cluster-upgrades.md

```diff
@@ -0,0 +1,221 @@
+	* [For CRDs](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#status-subresource): The `.metadata.generation` value is incremented for all changes, except for changes to `.metadata` or `.status`.
+* [`status.observedGeneration`](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties): is the `generation` most recently observed by the component responsible for acting upon changes to the desired state of the resource. In other words, the value of the `observedGeneration` field indicates the last time the controller updated the status subresource. This can be used, for instance, to ensure that the reported status reflects the most recent desired status.
+
+### Workflow: how to use `observedGeneration` to only reconcile on changes
+This diagram represents the logic to follow in the EKS-A `Cluster` controller:
+
+```mermaid
+stateDiagram-v2
+    [*] --> Queue: Cluster reconcile request
+    
+    deleted: Cluster deleted?
+    delete: Reconcile delete
+    reconcile: Reconcile cluster
+    observedGeneration: generation == observedGeneration
+    updateGeneration: observedGeneration = generation
+    done: Reconciliation done?
+
+    Queue --> deleted
+    state deleted <<choice>> 
+        deleted --> observedGeneration: no
+        deleted --> delete: yes
+    state observedGeneration <<choice>>
+        observedGeneration --> reconcile: no
+        observedGeneration --> [*]: yes
+    delete --> done
+    reconcile --> done
+    state done <<choice>>
+        done --> updateGeneration: yes
+        done --> Queue: no
+    updateGeneration  --> [*]
+```
+
+We only perform the actual reconciliation logic (this is trying to change the state of the world) when `observedGeneration` is different than `generation`. This will only happen after the resource has changed.
+
+It's important to mention that we only update the value of `observedGeneration` when the reconciliation process is "done". Done here means that the state of the world is equal to the desired state: the cluster has been completely reconciled. This is to take into account the case where a full reconciliation takes multiple loops (for example, when the cluster is first created). Note that "reconcile not done" also includes errors.
+
+This might be slightly different of how other resources use the `observedGeneration` field, where the field is always set on every reconciliation loop. For multi state resources (like our `Cluster`) this is usually accompanied of a "state" indicator (note that `phases` is now discouraged in favor of `conditions`). However, we don't have that in our `Cluster`'s status. Although tempting to try to add this to the status, we encourage to tackle this problem separately and revisit the use of `observedGeneration` if necessary. Refer to the [alternatives](#alternative-solutions) section for other possible solutions.
```

Comment:
I missed where you talk about using status in alternatives, unless that wasn't what you meant. Is it possible to talk about what is the additional effort required to use `observedGeneration` along with `ClusterStatus` because it seems like that might be the approach we want to go towards anyways.